source/jormungandr/jormungandr/street_network/geovelo.py

```diff
@@ -0,0 +1,190 @@
+        try:
+            return self.breaker.call(method, url, timeout=self.timeout, data=data, headers={'content-type': 'application/json', 'Api-Key': self.api_key})
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error('Geovelo routing service dead (error: {})'.format(e))
+            self.record_external_failure('circuit breaker open')
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error('Geovelo routing service dead (error: {})'.format(t))
+            self.record_external_failure('timeout')
+        except Exception as e:
+            logging.getLogger(__name__).exception('Geovelo routing error')
+            self.record_external_failure(str(e))
+        return None
+
+    def _get_matrix(self, json_response):
+        sn_routing_matrix = response_pb2.StreetNetworkRoutingMatrix()
+        row = sn_routing_matrix.rows.add()
+        for e in json_response:
+            if e[2] == 'duration':
+                continue
+            routing = row.routing_response.add()
+            if e[2]:
+                routing.duration = e[2]
+                routing.routing_status = response_pb2.reached
+            else:
+                routing.duration = -1
+                routing.routing_status = response_pb2.unknown
+        return sn_routing_matrix
+
+    def _check_response(self, response):
+        if response == None:
+            raise TechnicalError('impossible to access geovelo service')
+        if response.status_code != 200:
+            logging.getLogger(__name__).error('Geovelo service unavailable, impossible to query : {}'
+                                              ' with response : {}'
+                                              .format(response.url, response.text))
+            raise TechnicalError('Geovelo service unavailable, impossible to query : {}'.
+                                 format(response.url))
+
+    def get_street_network_routing_matrix(self, origins, destinations, street_network_mode, max_duration, request, **kwargs):
+        if street_network_mode != "bike":
+            logging.getLogger(__name__).error('Geovelo, mode {} not implemented'.format(street_network_mode))
+            raise InvalidArguments('Geovelo, mode {} not implemented'.format(street_network_mode))
+
+        data = self._make_data(origins, destinations)
+        r = self._call_geovelo('{}/{}'.format(self.service_url, 'api/v2/routes_m2m'),
+                               requests.post, json.dumps(data))
+        self._check_response(r)
+        resp_json = r.json()
+        return self._get_matrix(resp_json)
+
+    def _get_response(self, json_response, mode, pt_object_origin, pt_object_destination, datetime, clockwise):
+        resp = response_pb2.Response()
+        resp.status_code = 200
+        resp.response_type = response_pb2.ITINERARY_FOUND
+
+        duration = 13371337
```

Comment:
Note sure to understand this code.

there can be only one duration returned by geovelo ?

I think there should at least be more comments :wink: but why do you init this a this value ?